packages/workbox-core/models/messages/messages.mjs

```diff
@@ -32,6 +32,22 @@ export default {
       `'${moduleName}.${className}.${funcName}()' must be an array.`;
   },
 
+  'not-of-type': ({paramName, expectedType, moduleName, funcName}) => {
+    if (!paramName || !expectedType || !moduleName || !funcName) {
+      throw new Error(`Unexpected input to 'not-of-type' error.`);
+    }
+    return `The parameter '${paramName}' passed into ` +
+      `'${moduleName}.${funcName}()' must be of type ${expectedType}.`;
+  },
+
+  'not-a-method': ({paramName, expectedMethod, moduleName, funcName}) => {
+    if (!paramName || !expectedMethod || !moduleName || !funcName) {
+      throw new Error(`Unexpected input to 'not-a-method' error.`);
+    }
+    return `${moduleName}.${funcName}() was called with a ${paramName} that ` +
```

Comment:
For:

```
core.assert.hasMethod(handler, 'handle', {
    moduleName: 'workbox-routing',
    funcName: 'Route',
    paramName: 'handler',
});
```

Would result in:

```
workbox-routing.Route() was called with a handler that doesn't expose a handle method.
```

Could we tweak to:

```
workbox-routing.Router() expected the 'handler' parameter to expose a 'handle' method.
```

Reply:
Done.